File: src/Ritmax.Domain/Entities/Company.cs
namespace Ritmax.Domain.Entities;

public class Company
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? LegalName { get; set; }
    public string Gstin { get; set; } = string.Empty;
    public string Pan { get; set; } = string.Empty;
    public string? Cin { get; set; }
    public int? IndustryId { get; set; }
    public string Address { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public int? StateId { get; set; }
    public int? PinCode { get; set; }
    public string PhoneNo { get; set; } = string.Empty;
    public long? Mobile { get; set; }
    public string? Website { get; set; }
    public string? Email { get; set; }
    public string? LogoPath { get; set; }

    public bool IsDeleted { get; set; }
    public string? CreatedBy { get; set; }
    public DateTime CreatedDate { get; set; }
    public string? ModifiedBy { get; set; }
    public DateTime? ModifiedDate { get; set; }
    public string? UpdateTag { get; set; }
    public Guid UpdateGuid { get; set; }

    public Industry? Industry { get; set; }
    public State? State { get; set; }
    public ICollection<Department> Departments { get; set; } = [];
    public ICollection<PayrollRun> PayrollRuns { get; set; } = [];
}

File: client/ritmax-ui/src/shared/types/company.ts
export interface CompanyProfile {
  id: number
  name: string
  legalName?: string | null
  gstin: string
  pan: string
  cin?: string | null
  industryId?: number | null
  industryName?: string | null
  address: string
  city: string
  stateId?: number | null
  stateName?: string | null
  pinCode?: number | null
  phoneNo: string
  mobile?: number | null
  website?: string | null
  email?: string | null
  logoPath?: string | null
}

export interface Industry {
  id: number
  name: string
}

export interface StateOption {
  id: number
  name: string
}

export interface UpdateCompanyRequest {
  name: string
  legalName?: string | null
  gstin: string
  pan: string
  cin?: string | null
  industryId?: number | null
  address: string
  city: string
  stateId?: number | null
  pinCode?: number | null
  phoneNo: string
  mobile?: number | null
  website?: string | null
  email?: string | null
}
